devicetypes/smartthings/zwave-door-temp-sensor.src/zwave-door-temp-sensor.groovy

```diff
@@ -0,0 +1,311 @@
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ *  Z-Wave Door/Temp Sensor
+ *
+ */
+
+metadata {
+    definition (name: "Z-Wave Door/Temp Sensor", namespace: "smartthings", author: "SmartThings", ocfDeviceType: "x.com.st.d.sensor.contact", runLocally: true, minHubCoreVersion: '000.019.0012', executeCommandsLocally: false) {
+        capability "Contact Sensor"
+        capability "Sensor"
+        capability "Battery"
+        capability "Configuration"
+        capability "Health Check"
+        capability "Temperature Measurement"
+
+        fingerprint mfr:"015D", prod:"2003", model:"B41C", deviceJoinName: "Inovelli Door/Temp Sensor"
+        fingerprint mfr:"0312", prod:"2003", model:"C11C", deviceJoinName: "Inovelli Door/Temp Sensor"
+        fingerprint mfr:"015D", prod:"2003", model:"C11C", deviceJoinName: "Inovelli Door/Temp Sensor"
+        fingerprint mfr:"015D", prod:"C100", model:"C100", deviceJoinName: "Inovelli Door/Temp Sensor"
+        fingerprint mfr:"0312", prod:"C100", model:"C100", deviceJoinName: "Inovelli Door/Temp Sensor"
+    }
+
+// Temperature offset is not currently supported locally for the Z-Wave sensors
+// If needed then self-publish the DTH and uncomment the following code
+/*
+	preferences {
+		section {
+			input title: "Temperature Offset", description: "This feature allows you to correct any temperature variations by selecting an offset. Ex: If your sensor consistently reports a temp that's 5 degrees too warm, you'd enter '-5'. If 3 degrees too cold, enter '+3'.", displayDuringSetup: false, type: "paragraph", element: "paragraph"
+			input "tempOffset", "number", title: "Degrees", description: "Adjust temperature by this many degrees", range: "*..*", displayDuringSetup: false
+		}
+	}
+*/
+
+    tiles(scale: 2) {
+        multiAttributeTile(name:"contact", type: "generic", width: 6, height: 4){
+            tileAttribute ("device.contact", key: "PRIMARY_CONTROL") {
+                attributeState("open", label: '${name}', icon: "st.contact.contact.open", backgroundColor: "#e86d13")
+                attributeState("closed", label: '${name}', icon: "st.contact.contact.closed", backgroundColor: "#00A0DC")
+            }
+            tileAttribute("device.temperature", key: "SECONDARY_CONTROL") {
+                attributeState("default", label:'${currentValue}°',
+                        backgroundColors: [
+                                [value: 31, color: "#153591"],
+                                [value: 44, color: "#1e9cbb"],
+                                [value: 59, color: "#90d2a7"],
+                                [value: 74, color: "#44b621"],
+                                [value: 84, color: "#f1d801"],
+                                [value: 95, color: "#d04e00"],
+                                [value: 96, color: "#bc2323"]
```

Comment:
we often do essentially two ranges of temperatures/colors here for C/F.

Reply:
Also I'm not sure exactly how this color is displayed in a multiattribute tile, you may want to check.

Reply:
Okay. I figured it would just get ignored. You can get rid of the `backgroundColor` specifications for the temperature attribute. Since it's just a secondary attribute of the open/closed tile, those color values are ignored.